### PR TITLE
Backend sprint3

### PR DESCRIPTION
**Pull request by xXEmilioXx**

### CODE REVIEW

File: backendStub/service/FrikkService.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    "tags" : [ {
+      "name" : "Mediterranean",
+      "color" : "#FFBC8C"
+    }, {
+      "name" : "Pasta",
+      "color" : "#99C99B"
+    }, {
+      "name" : "Italian",
+      "color" : "#FFBC8C"
+    } ],
```
Comment (cathrbak) [inline backendStub/service/FrikkService.js]: I am still missing the menuItemId in this menuItem

File: backendStub/service/FrikkService.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  return new Promise(function(resolve, reject) {
+    var examples = {};
+    examples['application/json'] = {
+  "restaurant" : {
+    "restaurantName" : "Emilio's Pizza",
+    "address" : "André route 45",
+    "city" : "Rome",
+    "country" : "Italy",
+    "latitude" : 48.4281350802915,
+    "longitude" : -110.0792542197085
```
Comment (cathrbak) [inline backendStub/service/FrikkService.js]: Can we change this to: 

`"address" : " Piazzale Susa",`
 `   "city" : "Milan",`
  `  "country" : "Italy",`
  `  "latitude" : 45.4688346,`
  `  "longitude" : 9.2234114`

Makes no sense to test functionality with random data

File: backendStub/service/FrikkService.js (hunk 1 above)
Comment (cathrbak) [inline backendStub/service/FrikkService.js]: Can you add this tag as well? We would like to test the maximum number of tags which is 4. 
`,
      {
        "name" : "Dinner",
        "color" : "#99C99B"
      }`

File: backendStub/service/FrikkService.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+ * Return all the menus of the restaurant
+ * Return all the menus of the restaurant. Since authentication is required, the backend is able to get which restaurant is involved from the authentication token.
+ *
+ * returns List
+ **/
+exports.getOwnMenus = function() {
+  return new Promise(function(resolve, reject) {
+    var examples = {};
+    examples['application/json'] = [ {
+  "menuID" : 2,
```
Comment (cathrbak) [inline backendStub/service/FrikkService.js]: The two menus returned here have the same ID, can this be changed? It violates the unique item constraints of menus

File: backendStub/service/FrikkService.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+exports.searchByMenuItem = function(menuItemName) {
+  return new Promise(function(resolve, reject) {
+    var examples = {};
+    examples['application/json'] = [ {
+  "restaurantData" : {
+    "restaurantName" : "Emilio's Pizza",
+    "restaurantImageLink" : "www.cloudStorage.com/Restaurant"
+  },
+  "menu" : {
+    "menuID" : 2,
```
Comment (cathrbak) [inline backendStub/service/FrikkService.js]: Same non-unique menuID here

File: routes/searchMenu.js [hunk 5]
```diff
@@ -7,53 +7,108 @@ const Menu = require('../models/menu.js');
...
                 priceEuros: menuItems.dataValues.Price,
                 imageLink: menuItems.dataValues.ImageLink,
-                tags}
+                rating: average,
+                tags
+            }
         });
         const menuItems = await Promise.all(promises);
 
         res.status(200).send({
```
Comment (cathrbak) [inline routes/searchMenu.js]: Maybe I'm just blind, but I can find the response I will get if no menuItems are found?

Comment (Frikkha) [reply routes/searchMenu.js]: You are right, now you just get an empty response together with the 400 status. Will fix it
